src/plugins/intel_cpu/CMakeLists.txt

```diff
@@ -20,18 +20,33 @@ elseif(OV_COMPILER_IS_CLANG)
     ie_add_compiler_flags(-Wno-delete-non-abstract-non-virtual-dtor)
 endif()
 
+set(OV_CPU_ARM_TARGET_GENERIC_ARCHS armv8a
+                                    armv8.2-a
+                                    armv8.6-a armv8.6-a-sve armv8.6-a-sve2 armv8.6-a-sve2-sme2
+                                    armv8r64 # the same as armv8.4-a
+)
 if(ARM)
     set(OV_CPU_ARM_TARGET_ARCH_DEFAULT armv7a)
+    set(OV_CPU_ARM_TARGET_ARCHS armv7a armv7a-hf
+                                # requires estate=32
+                                ${OV_CPU_ARM_TARGET_GENERIC_ARCHS})
 else()
     if(APPLE)
         # Apple M1 / M2 is assumed
         set(OV_CPU_ARM_TARGET_ARCH_DEFAULT arm64-v8.2-a)
-        add_definitions(-DOV_CPU_ARM_ENABLE_FP16)
     else()
         set(OV_CPU_ARM_TARGET_ARCH_DEFAULT arm64-v8a)
     endif()
+    set(OV_CPU_ARM_TARGET_ARCHS arm64-v8a
+                                arm64-v8.2-a arm64-v8.2-a-sve arm64-v8.2-a-sve2
+                                # used with estate=64
+                                ${OV_CPU_ARM_TARGET_GENERIC_ARCHS})
+endif()
+set(OV_CPU_ARM_TARGET_ARCH ${OV_CPU_ARM_TARGET_ARCH_DEFAULT} CACHE STRING "Architecture for ARM ComputeLibrary")
+set_property(CACHE OV_CPU_ARM_TARGET_ARCH PROPERTY STRINGS ${OV_CPU_ARM_TARGET_ARCHS})
+if(OV_CPU_ARM_TARGET_ARCH MATCHES "arm64-v8.2-a")
```

Comment:
@dmitry-gorokhov should we enable `OV_CPU_ARM_ENABLE_FP16` for Apple architecture (arm64-v8.2-a) only? Or should we extend this condition to enable FP16 on any arm64 architecture?

Reply:
with such hardcoded condition you are missing other archs where FP16 can be enabled. We need to detect all ways where FP16 can be enabled 

as far as I can see, all arches starting with arm64-v8.2 implicitly or explicitly enable fp16.

Reply:
BTW, returning back to this comment https://github.com/openvinotoolkit/openvino/pull/18394#discussion_r1278840069

Maybe we need to have something like:
```c++
bool mayiuse_arm64_fp16() {
#ifdef__ARM_FEATURE_FP16_VECTOR_ARITHMETIC
    return true;
#else
    return false;
#endif
}
```
And this is HW capability to use FP16 kernels, while product-level FP16 is determined by plugin (e.g. inference_precision is set to FP32 explicitly)

In this case we don't have to guess in cmake whether FP16 is enabled.

Reply:
@ilya-lavrenov I had a talk with Dmitry, he has decided to keep the current implementation since `__ARM_FEATURE_FP16_VECTOR_ARITHMETIC` can't be checked in cmake file.

Reply:
Please, restore ability to select arch via cmake.